### PR TITLE
Download nativeScript-migration-data.json file

### PR DESCRIPTION
**Pull request by TsvetanMilanov**
When we integrate experimental version of NativeScript we need to upload new package of the CLI in npm with the updated nativeScript-migration-data.json file so the users can use the new version.

It will be better to download the nativeScript-migration-data.json file when NativeScript related command is executed.

### CODE REVIEW

File: lib/project/nativescript-project.ts [hunk 1]
```diff
@@ -62,6 +67,22 @@ export class NativeScriptProject extends FrameworkProjectBase implements Project
 		return [this.$projectConstants.PACKAGE_JSON_NAME];
 	}
 
+	private get $nativeScriptMigrationService(): IFrameworkMigrationService {
+		if (!this._nativeScriptMigrationService) {
+			this._nativeScriptMigrationService = this.$injector.resolve("nativeScriptMigrationService");
+		}
+
+		return this._nativeScriptMigrationService;
```
Comment (rosen-vladimirov) [inline lib/project/nativescript-project.ts]: there's no need to cache this, the `$injector` is caching the dependencies

File: lib/project/nativescript-project.ts [hunk 2]
```diff
@@ -62,6 +67,22 @@ export class NativeScriptProject extends FrameworkProjectBase implements Project
...
+	}
+
+	private get $nativeScriptResources(): INativeScriptResources {
+		if (!this._nativeScriptResources) {
```
Comment (rosen-vladimirov) [inline lib/project/nativescript-project.ts]: there's no need to cache this, the `$injector` is caching the dependencies

File: lib/project.ts [hunk 3]
```diff
@@ -51,6 +51,11 @@ export class Project extends ProjectBase implements Project.IProject {
 				"The AppBuilder CLI lets you target only Apache Cordova 3.0.0 or later. " +
 				"To develop your projects with Apache Cordova 2.x, run the AppBuilder Windows client or the in-browser client.");
 		}
+
+		if (this.projectData && this.projectData.Framework) {
+			this.frameworkProject = this.$frameworkProjectResolver.resolve(this.projectData.Framework);
```
Comment (rosen-vladimirov) [inline lib/project.ts]: we must make `frameworkProject` to has only getter, no need to set it on several places in this file. However this should be handled in separate PR

File: lib/project/nativescript-project.ts [hunk 4]
```diff
@@ -144,5 +165,39 @@ export class NativeScriptProject extends FrameworkProjectBase implements Project
...
+
+			try {
+				this.$nativeScriptMigrationService.downloadMigrationConfigFile(downloadedFilePath).wait();
+			} catch (err) {
+				// This exception is caused probably by issue with the internet connection of the user.
+				this.$logger.trace("Failed to download the NativeScript migration file.");
+				return;
+			}
+
+			let newMigrationFileContent = this.$fs.readFile(downloadedFilePath).wait().toString();
```
Comment (rosen-vladimirov) [inline lib/project/nativescript-project.ts]: you can use:
```TypeScript
this.$fs.readText(downloadedFilePath).wait();
```
and skip `.toString()`

File: lib/project/nativescript-project.ts [hunk 5]
```diff
@@ -144,5 +165,39 @@ export class NativeScriptProject extends FrameworkProjectBase implements Project
...
+			let currentMigrationFileContent = this.$fs.readFile(nativeScriptMigrationFileName).wait().toString();
```
Comment (rosen-vladimirov) [inline lib/project/nativescript-project.ts]: same as above

**Comment by TsvetanMilanov** [comment]
run ci

**Comment by TsvetanMilanov** [comment]
run ci

**Comment by rosen-vladimirov** [comment]
:+1: great work

**Comment by TsvetanMilanov** [comment]
run ci

**Comment by TsvetanMilanov** [comment]
run ci
